assets/js/lesson_20/students_vs_teacher.js

```diff
@@ -0,0 +1,149 @@
+class Human {
+    constructor(obj) {
+        this.name = obj.name;
+        this.surname = obj.surname;
+        this.age = obj.age;
+    }
+
+    getFullName() {
+        return `${this.name} ${this.surname}`;
+    }
+
+    setFullName(fullName) {
+        [this.name, this.surname] = fullName.split(' ');
+    }
+}
+
+class Student extends Human {
+    constructor(info) {
+        super(info);
+        this.marks = info.marks;
+    }
+
+    getAverageMark() {
+        return this.marks.reduce((a, b) => a + b) / this.marks.length;
+    }
+
+    getMinMark() {
+        return Math.min(...this.marks);
+    }
+
+    getMaxMark() {
+        return Math.max(...this.marks);
+    }
+}
+
+class Teacher extends Human {
+    constructor(info) {
+        super(info);
+        this.group = info.group;
+    }
+
+    getListOfNamesByAverageMark() {
+        return this.group.sort((a, b) => b.getAverageMark() - a.getAverageMark()).map(student => student.getFullName());
+    }
+
+    getStudentByName(name) {
+        return this.group.find(student => student.getFullName() === name);
+    }
+
+    removeStudentByName(name) {
+        this.group = this.group.filter(student => student.getFullName() !== name);
+    }
+
+    updateStudentByName(newStudent, name) {
+        let index = this.group.findIndex(student => student.getFullName() === name);
+        if (index !== -1) {
+            this.group[index] = newStudent;
+        }
+    }
+}
```

Comment:
In the local scope of Teacher class, you should implement findFakeStudent method. Which returns string with the name of FakeStudent and his real marks.